### PR TITLE
Judicial-system/Assign registrar

### PR DESCRIPTION
**Pull request by oddsson**
# Assign registrar

https://app.asana.com/0/1199153462262248/1199174045356610

## What

Add ability to assign registrars to a case.

## Why

It's a requirement.

## Screenshots / Gifs

screencapture-localhost-4200-domur-fyrirtokutimi-4cc064ae-6096-4731-9db1-539188ee24bf-2021-02-11-15_42_09

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/web/src/routes/Judge/CourtRecord/CourtRecord.tsx [hunk 1]
```diff
@@ -107,6 +107,10 @@ export const CourtRecord: React.FC = () => {
       if (wc.judge) {
         attendees += `${wc.judge.name} ${wc.judge.title}\n`
       }
+      console.log(wc)
```
Comment (oddsson) [inline apps/judicial-system/web/src/routes/Judge/CourtRecord/CourtRecord.tsx:110]:
```suggestion

```

File: apps/judicial-system/backend/src/app/formatters/pdf.ts [hunk 2]
```diff
@@ -453,6 +453,17 @@ export async function generateRulingPdf(existingCase: Case): Promise<string> {
     )
     .text(' ')
     .font('Helvetica-Bold')
+    .text(
+      `${
+        existingCase.registrar?.name || 'Dómritari hefur ekki verið skráður'
+      } ${existingCase.registrar?.title || ''}`,
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/formatters/pdf.ts:459]: According to the Asana task, the registrar should not show up here. This is the signature line of the judge.

The registrar should be autofilled in the courtAttendees field in the client. That field is already in the pdf, so no need to add anything here. See:
image
image

File: apps/judicial-system/web/src/routes/Judge/CourtRecord/CourtRecord.tsx [hunk 3]
```diff
@@ -108,6 +108,10 @@ export const CourtRecord: React.FC = () => {
         attendees += `${wc.judge.name} ${wc.judge.title}\n`
       }
 
+      if (wc.registrar) {
+        attendees += `${wc.registrar.name} ${wc.registrar.title}\n`
+      }
+
```
Comment (gudjong) [inline apps/judicial-system/web/src/routes/Judge/CourtRecord/CourtRecord.tsx:114]: This is all that is needed for the registrar to show up in the pdf. See earlier comment.

File: apps/judicial-system/web/src/routes/Judge/HearingArrangements/HearingArrangements.tsx [hunk 4]
```diff
@@ -239,6 +272,27 @@ export const HearingArrangements: React.FC = () => {
               required
             />
           </Box>
+          <Box component="section" marginBottom={5}>
+            <Box marginBottom={3}>
+              <Text as="h3" variant="h3">
+                Dómritari{' '}
+                <Tooltip text="Dómritari sem er valinn hér verður skráður á málið og mun fá tilkynningar sendar í tölvupósti." />
```
Comment (gudjong) [inline apps/judicial-system/web/src/routes/Judge/HearingArrangements/HearingArrangements.tsx:279]: Have email notifications to the registrar been implemented?

**Review by gudjong — CHANGES_REQUESTED** [review]

File: apps/judicial-system/backend/src/app/formatters/pdf.ts (hunk 2 above)
Comment (oddsson) [reply apps/judicial-system/backend/src/app/formatters/pdf.ts:459]: Good eye! Thank you 🙏

**Review by gudjong — APPROVED** [review]

**Pull request merged**